target/scripts/startup/setup.d/mta_sts.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# Set up MTA-STS
+
+function _setup_mta_sts() {
+  _log 'trace' 'Adding MTA-STS lookup to the Postfix TLS policy map'
+  postconf 'smtp_tls_policy_maps = socketmap:inet:127.0.0.1:8461:postfix'
```

Comment:
It's messy as it requires inter-dependencies between containers. For something that I would consider an important security property, this problematic. I don't have a strong preference for communication locally via TCP oder UDS.

I'm still looking into the best option for testing. It seems to be a pity that there currently is no easy test domain in the wild and I'm trying to reach out to folks in the hope of something being willing to host one.

Reply:
It's not too difficult to do this all locally, we don't need an external test domain.

It's just something I've been putting off until the test suite is refactored more to switch to docker compose as I can more easily manage flexible configurations.

---

If you'd like to implement it, I'll allow that.
- Use a [Caddy container](https://hub.docker.com/_/caddy) to host the policy file. We can use our [existing local testing certs](https://github.com/docker-mailserver/docker-mailserver/tree/master/test/test-files/ssl/example.test), Caddy needs to be provided these along with a little config.
- Use a [CoreDNS container](https://hub.docker.com/r/coredns/coredns/) with a BIND file zone to handle the DNS records.
- Ensure that DMS has the CA root cert in the trust store for verifying trust against the Caddy container. The query CLI tool from the project your using packages that separately on Debian.
- I'm not sure if submitting mail to be sent to a domain the DMS container manages itself will work for the logs, it might optimize that delivery if it recognizes that it loops back into itself. That may require another DMS instance, but I don't think we're setup for that supporting that in tests yet 🤔(maybe)

I have familiarity with all of the above, so could probably contribute the Caddy + CoreDNS setup via a separate PR, then you could rebase to leverage that. I'll let you know if I get that handled, if you know of a better way I'm open to ideas :)

Reply:
The main reason I would like to see a test domain exist is to give the user a way to test the correct behavior on their own. E.g. for DKIM there are validation services and the report functions. Without this testing, #3593 might have gone undetected for a while. I was just trying to resist the setup so far :)

Reply:
I have a working local offline setup with caddy + coredns to resolve the lookup via `mta-sts-query` from the package. Uses the test CA + cert I linked for TLS.

A user could setup the same locally with two DMS instances to test setup is correct, then after that is successful use real public DNS + certs 🤷‍♂️ 

I'll put a test together tomorrow to handle the sending and logs verification.

---

Our test suite is slowly getting better, once it adopts the local DNS setup more broadly it can have better test coverage to catch issues like you've linked, assuming contributors have time to write the tests 😅 

> The main reason I would like to see a test domain exist is to give the user a way to test the correct behavior on their own.
> E.g. for DKIM there are validation services and the report functions.

I can't assist much with public validation concern, but I think being able to have an offline setup is a useful tool for getting confidence in getting everything working for new users :)